core/app/presenters/spree/filters/property.rb

```diff
@@ -0,0 +1,20 @@
+module Spree
+  module Filters
+    class Property
+      def initialize(property:, product_properties:)
+        @property = property
+        @product_properties = product_properties
+      end
+
+      delegate_missing_to :property
+
+      def uniq_values
```

Comment:
@mad-eel I think we should use this method - https://github.com/spree/spree/blob/master/core/app/models/spree/property.rb#L22 - 

do we even use this `uniq_values` method anywhere?

Reply:
Yes. For filtering, we first fetch product properties and then group them by property so we can access them on the view. But these are not all product properties and the method on the property model fetches unique values for all of its product properties. 

This method (overriding the original one) is used when displaying product properties on the filters panel: https://github.com/spree/spree/blob/79ea3d26e8844a01544f9fa3ab033c567c83184c/frontend/app/views/spree/shared/_properties.html.erb#L2

And this presenter is used when building the list for `available_properties` here: https://github.com/spree/spree/blob/876da2052248c7be557d93795ac98b173ad1fb85/frontend/app/helpers/spree/frontend_helper.rb#L314

Alternatively, we can adjust the method on the Property model so we can pass a scope for product properties there. What do you think?

Reply:
I made that change, let me know what you think: https://github.com/spree/spree/pull/11023/files#diff-f26177b5df7f418dc85dd8c4ea874b6289b73cf36654b241f57fd4ee74109577R22

Reply:
@damianlegawiec I made that change, let me know what you think: https://github.com/spree/spree/pull/11023/files#diff-f26177b5df7f418dc85dd8c4ea874b6289b73cf36654b241f57fd4ee74109577R22